packages/medusa/src/loaders/helpers/register-workflows.ts

```diff
@@ -1,21 +1,20 @@
-import { PluginDetails } from "@medusajs/types"
 import { glob } from "glob"
-import { getResolvedPlugins } from "./resolve-plugins"
+import { PluginDetails } from "@medusajs/types"
 
 /**
  * import files from the workflows directory to run the registration of the wofklows
  * @param pluginDetails
  */
-async function registerWorkflows(pluginDetails: PluginDetails): Promise<void> {
-  const files = glob.sync(`${pluginDetails.resolve}/workflows/*.js`, {})
-  await Promise.all(files.map(async (file) => import(file)))
-}
-
-export async function registerProjectWorkflows({
-  rootDirectory,
-  configModule,
-}) {
-  const [resolved] =
-    getResolvedPlugins(rootDirectory, configModule, "dist", true) || []
-  await registerWorkflows(resolved)
+export async function registerWorkflows(
+  plugins: PluginDetails[]
+): Promise<void> {
+  await Promise.all(
+    plugins.map(async (pluginDetails) => {
+      const files = glob.sync(
+        `${pluginDetails.resolve}/workflows/*(.js|.ts|.mjs)`,
```

Comment:
suggestion: I believe we should exclude definition and map file wdyt?

Reply:
Q: Does the resolve include the dist or src?

Reply:
I think barrel files as well in the exclusion

Reply:
> Q: Does the resolve include the dist or src?

I think it will be both. In development it will be from the `src` folder and in prod it will be from the compiled output, ie `dist`.